plutus-playground-client/src/Chain/View.purs

```diff
@@ -0,0 +1,292 @@
+          , cardBody_ [ valueView value ]
+          ]
+      ]
+
+showSequenceId :: forall p i. SequenceId -> HTML p i
+showSequenceId (SequenceId { slotIndex, txIndex }) =
+  span_
+    [ text "Slot #"
+    , text $ show slotIndex
+    , text ", "
+    , text "Transaction #"
+    , text $ show txIndex
+    ]
+
+txOutOfView :: forall p. Boolean -> Map PubKey Wallet -> TxOutOf String -> HTML p (Query Unit)
+txOutOfView showArrow walletKeys txOutOf@(TxOutOf { txOutAddress, txOutType, txOutValue }) =
+  div
+    [ classes [ card, entry, outTypeClass txOutType ]
+    , onClick
+        $ const
+        $ Just
+        $ action
+        $ SetChainFocus
+        $ Just
+        $ FocusAddress txOutAddress
+    ]
+    [ cardHeaderTxOutTypeView showArrow walletKeys txOutType
+    , cardBody_
+        [ valueView txOutValue ]
+    ]
+
+outTypeClass :: TxOutType -> ClassName
+outTypeClass (PayToPubKey _) = ClassName "wallet"
+
+outTypeClass (PayToScript _) = ClassName "script"
+
+cardHeaderTxOutTypeView :: forall p i. Boolean -> Map PubKey Wallet -> TxOutType -> HTML p i
+cardHeaderTxOutTypeView showArrow walletKeys txOutType =
+  div [ classes [ cardHeader, textTruncate ] ]
+    [ if showArrow then triangleRight else text ""
+    , txOutTypeView walletKeys txOutType
+    ]
+
+txOutTypeView :: forall p i. Map PubKey Wallet -> TxOutType -> HTML p i
+txOutTypeView walletKeys (PayToPubKey pubKey) = case Map.lookup pubKey walletKeys of
+  Nothing -> showPubKey pubKey
+  Just (Wallet { getWallet: n }) ->
+    span_
+      [ showPubKey pubKey
+      , br_
+      , small_
+          [ text "Wallet"
+          , nbsp
+          , text $ show n
+          ]
+      ]
+
+txOutTypeView _ (PayToScript dataScript) = showDataScript dataScript
+
+showDataScript :: forall p i. DataScript -> HTML p i
+showDataScript (DataScript { getDataScript: d }) =
+  span_
+    [ text "DataScript"
+    , nbsp
+    , text d
+    ]
+
+showPubKey :: forall p i. PubKey -> HTML p i
+showPubKey (PubKey { getPubKey: p }) =
+  span_
+    [ text "PubKey"
+    , nbsp
+    , text p
+    ]
+
+valueView :: forall p i. Value -> HTML p i
+valueView (Value { getValue: (AssocMap.Map []) }) = empty
+
+valueView (Value { getValue: (AssocMap.Map currencies) }) = div_ (currencyView <$> currencies)
+  where
+  currencyView :: JsonTuple CurrencySymbol (AssocMap.Map TokenName Int) -> HTML p i
+  currencyView (JsonTuple (currency /\ (AssocMap.Map tokens))) =
+    row_
+      [ col4_ [ text $ showCurrency currency ]
+      , col_ (tokenView <$> tokens)
+      ]
+
+  tokenView :: JsonTuple TokenName Int -> HTML p i
+  tokenView (JsonTuple (token /\ balance)) =
+    div_
+      [ text $ showToken token
+      , text ": "
+      , text $ show balance
+      ]
+
+  showCurrency (CurrencySymbol { unCurrencySymbol: "" }) = "Ada"
+
+  showCurrency (CurrencySymbol { unCurrencySymbol: symbol }) = symbol
+
+  showToken (TokenName { unTokenName: "" }) = "Lovelace"
+
+  showToken (TokenName { unTokenName: name }) = name
```

Comment:
OT: it really seems wrong to have an empty currency type

Reply:
I kinda feel the same. But that's one for @j-mueller...